### DISCUSSION TITLE
Use of I2S_NUM_1

### DISCUSSION THREAD

**Discussion by Taragorm** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#203] [Q&A] [answered]
Is it possible to use a different I2S device?

I have an application that needs DMA to the DACs, and I understand that requires I2S0.

I had a poke through the source, and while various port parameters exist, they are mostly overridden and therefore ineffective. Is there a reason for this?

**Comment by mrcodetastic** [comment]
Yes. Just do a search replace in ESP32-HUB75-MatrixPanel-I2S-DMA.cpp for I2S_NUM_0 and change it to I2S_NUM_1

There are two lines.

**Reply by mrcodetastic** [reply]
I2S_NUM_0 is used by default so it works across all ESP32 variants (like S2, S3) which only have one I2S device.

**Reply by Taragorm** [reply]
That was quick!
Thanks.

**Comment by mrcodetastic** [comment, marked as answer]
I have made a recent commit to be able to configure this in the main .h file of the library now.

https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-I2S-DMA/commit/3bd87dcfd69065acb4b5fd02403f8717ec8288d7
